packages/core/components/RangePicker/index.tsx

```diff
@@ -0,0 +1,171 @@
+import { ActionButton, Spinner, SpinnerSize } from "@fluentui/react";
+import classNames from "classnames";
+import * as React from "react";
+import FileFilter from "../../entity/FileFilter";
+import { extractValuesFromRangeOperatorFilterString } from "../../entity/AnnotationFormatter/number-formatter";
+import { AnnotationValue } from "../../services/AnnotationService";
+
+import styles from "./RangePicker.module.css";
+
+export interface ListItem {
+    displayValue: AnnotationValue;
+    value: AnnotationValue;
+}
+
+interface RangePickerProps {
+    disabled?: boolean;
+    errorMessage?: string;
+    items: ListItem[];
+    loading?: boolean;
+    onSearch: (filterValue: string) => void;
+    onReset: () => void;
+    currentRange: FileFilter | undefined;
+    units?: string;
+}
+
+/**
+ * A RangePicker is a simple form that renders input fields for the minimum and maximum values
+ * desired by the user and buttons to submit and reset the range. It also displays the
+ * overall min and max for that annotation, if available, and allows a user to select that full range.
+ *
+ * It is best suited for selecting items that are numbers.
+ */
+export default function RangePicker(props: RangePickerProps) {
+    const { errorMessage, items, loading, onSearch, onReset, currentRange, units } = props;
+
+    const overallMin = React.useMemo(() => {
+        return items[0]?.displayValue.toString() ?? "";
+    }, [items]);
+    const overallMax = React.useMemo(() => {
+        return items.at(-1)?.displayValue.toString() ?? "";
```

Comment:
Should this be +1 due to max being exclusive?

Reply:
re meeting yesterday, leaving this exclusive and making it clearer in UI that it is